core/src/test/java/org/apache/iceberg/data/DataTest.java

```diff
@@ -129,28 +135,37 @@ protected boolean supportsVariant() {
     return false;
   }
 
+  protected boolean supportsGeospatial() {
+    return false;
+  }
+
   @ParameterizedTest
   @FieldSource("SIMPLE_TYPES")
   public void testTypeSchema(Type type) throws IOException {
-    Assumptions.assumeThat(
-            supportsUnknown()
-                || TypeUtil.find(type, t -> t.typeId() == Type.TypeID.UNKNOWN) == null)
-        .as("unknown is not yet implemented")
-        .isTrue();
-    Assumptions.assumeThat(
-            supportsTimestampNanos()
-                || TypeUtil.find(type, t -> t.typeId() == Type.TypeID.TIMESTAMP_NANO) == null)
-        .as("timestamp_ns is not yet implemented")
-        .isTrue();
-    Assumptions.assumeThat(
-            supportsVariant()
-                || TypeUtil.find(type, t -> t.typeId() == Type.TypeID.VARIANT) == null)
-        .as("variant is not yet implemented")
-        .isTrue();
+    if (!supportsUnknown()) {
+      assumeNoUnsupportedType(type, Type.TypeID.UNKNOWN, "unknown");
+    }
+    if (!supportsTimestampNanos()) {
+      assumeNoUnsupportedType(type, Type.TypeID.TIMESTAMP_NANO, "timestamp_ns");
+    }
+    if (!supportsVariant()) {
+      assumeNoUnsupportedType(type, Type.TypeID.VARIANT, "variant");
+    }
+    if (!supportsGeospatial()) {
+      assumeNoUnsupportedType(type, Type.TypeID.GEOMETRY, "geometry");
+      assumeNoUnsupportedType(type, Type.TypeID.GEOGRAPHY, "geography");
+    }
 
     writeAndValidate(new Schema(required(1, "id", LongType.get()), optional(2, "test_type", type)));
   }
 
+  private static void assumeNoUnsupportedType(
```

Comment:
I'd rather not refactor this out since it is a method added with just one statement.